marc_to_solr/lib/process_holdings_helpers.rb

```diff
@@ -73,54 +78,35 @@ def in_temporary_location(field_876, field_852)
     current_location != 'RES_SHARE$IN_RS_REQ' && current_location != permanent_location_code(field_852)
   end
 
-  # Build the current (temporary) holding.
-  def current_holding(holding_current, field_852, field_876)
-    holding_current["location_code"] ||= current_location_code(field_876)
-    holding_current['current_location'] ||= Traject::TranslationMap.new("locations", default: "__passthrough__")[holding_current['location_code']]
-    holding_current['current_library'] ||= Traject::TranslationMap.new("location_display", default: "__passthrough__")[holding_current['location_code']]
-    holding_current['call_number'] ||= []
-    holding_current['call_number'] << [field_852['h'], field_852['i'], field_852['k'], field_852['j']].compact.reject(&:empty?)
-    holding_current['call_number'].flatten!
-    holding_current['call_number'] = holding_current['call_number'].join(' ').strip if holding_current['call_number'].present?
-    holding_current['call_number_browse'] ||= []
-    holding_current['call_number_browse'] << [field_852['h'], field_852['i'], field_852['k'], field_852['j']].compact.reject(&:empty?)
-    holding_current['call_number_browse'].flatten!
-    holding_current['call_number_browse'] = holding_current['call_number_browse'].join(' ').strip if holding_current['call_number_browse'].present?
-    # Updates current holding key; values are from 852
-    if field_852['l']
-      holding_current['shelving_title'] ||= []
-      holding_current['shelving_title'] << field_852['l']
-    end
-    if field_852['z']
-      holding_current['location_note'] ||= []
-      holding_current['location_note'] << field_852['z']
+  def build_call_number(field_852)
+    call_number = []
+    call_number << [field_852['h'], field_852['i'], field_852['k'], field_852['j']].compact.reject(&:empty?)
```

Comment:
can these two lines be combined?

Also, would `reject(&:blank?)` have the same effect as `.compact.reject(&:empty?)`

Reply:
Yup and Yup! And we can get rid of the `.flatten!`